aws/resource_aws_kinesis_analytics_application_test.go

```diff
@@ -567,6 +589,93 @@ resource "aws_kinesis_analytics_application" "test" {
 `, rInt, streamName, rInt, rInt)
 }
 
+func testAccKinesisAnalyticsApplication_inputsKinesisFirehose(rInt int) string {
+	return fmt.Sprintf(`
+data "aws_iam_policy_document" "trust_firehose" {
+  statement = {
+    actions = ["sts:AssumeRole"]
+    principals = {
+      type = "Service"
+      identifiers = ["firehose.amazonaws.com"]
+    }
+  }
+}
+
+resource "aws_iam_role" "firehose" {
+  name = "testAcc-firehose-%d"
+  assume_role_policy = "${data.aws_iam_policy_document.trust_firehose.json}"
+}
+
+data "aws_iam_policy_document" "trust_lambda" {
+  statement = {
+    actions = ["sts:AssumeRole"]
+    principals = {
+      type = "Service"
+      identifiers = ["lambda.amazonaws.com"]
+    }
+  }
+}
+
+resource "aws_iam_role" "lambda" {
+  name = "testAcc-lambda-%d"
+  assume_role_policy = "${data.aws_iam_policy_document.trust_lambda.json}"
+}
+
+resource "aws_s3_bucket" "test" {
+  bucket = "testacc-%d"
+  acl = "private"
+}
+
+resource "aws_lambda_function" "test" {
+  filename      = "test-fixtures/lambdatest.zip"
+  function_name = "testAcc-%d"
+  handler       = "exports.example"
+  role          = "${aws_iam_role.lambda.arn}"
+  runtime       = "nodejs8.10"
+}
+
+resource "aws_kinesis_firehose_delivery_stream" "test" {
+  name = "testAcc-%d"
+  destination = "extended_s3" 
+  extended_s3_configuration = {
+    role_arn = "${aws_iam_role.firehose.arn}"
+    bucket_arn = "${aws_s3_bucket.test.arn}"
+  }
+}
+
+resource "aws_kinesis_analytics_application" "test" {
+  name = "testAcc-%d"
+  code = "testCode\n"
+  inputs = {
```

Comment:
@kl4w FYI in Terraform 0.12, this syntax is no longer valid for configuration blocks and the configuration upgrade tool will fix it to `input {`

(same with other configuration blocks in this test configuration)